File: occasum-core/src/main/java/pw/stamina/occasum/properties/primitives/clamp/RangeIntClamp.java
package pw.stamina.occasum.properties.primitives.clamp;

final class RangeIntClamp implements IntClamp {
    private final int min, max;

    RangeIntClamp(int min, int max) {
        this.min = min;
        this.max = max;
    }

    @Override
    public int clamp(int value) {
        if (value < min) {
            return min;
        } else if (value > max) {
            return max;
        } else {
            return value;
        }
    }
}

File: occasum-core/src/main/kotlin/pw/stamina/occasum/properties/primitives/clamp/RangeIntClamp.kt
package pw.stamina.occasum.properties.primitives.clamp

internal class RangeIntClamp(private val min: Int, private val max: Int) : IntClamp {

    override fun clamp(value: Int): Int {
        return when {
            value < min -> min
            value > max -> max
            else -> value
        }
    }
}
